src/main/java/com/fulcrumgenomics/util/PickLongIndices.scala

```diff
@@ -0,0 +1,291 @@
+  @arg(          doc="The lowest acceptable secondary structure deltaG.")              val minDeltaG: Double = -10,
+  @arg(          doc="Adapter sequence(s) indices will be inserted into.", minElements=0) val adapters: Seq[String] = Seq(),
+  @arg(          doc="Sequences that should be avoided.  Any kmer of 'length' that appears in these sequences and their " + "reverse complements will be thrown out.")
+  val avoidSequence: Seq[String] = IlluminaAdapters.all.flatMap(_.both)
+) extends FgBioTool with LazyLogging {
+  // A few "constants"
+  private val Bases  = Array[Byte]('A', 'C', 'G', 'T')
+  private val random = new Random(seed)
+
+  // Input checking
+  existing.foreach(Io.assertReadable)
+  Io.assertCanWriteFile(output)
+  if (length < 1)                invalid("Length must be a positive integer.")
+  if (numberOfIndices < 1)       invalid("Number of indices to pick must be a positive integer.")
+  if (maxHomopolymer < 0)        invalid("Max homopolymer length must be >= 0.")
+  if (minGc >= maxGc)            invalid("Min GC must be greater than max GC.")
+  if (editDistance < 0)          invalid("Edit distance must be >= 0.")
+  if (editDistance > length)     invalid("Edit distance must be <= length.")
+  if (viennaRnaDir.isDefined && adapters.isEmpty) invalid("Specifying Vienna RNA dir has no effect if no adapters are provided.")
+  if (viennaRnaDir.isEmpty && adapters.nonEmpty)  invalid("Specifying adapters has no effect if Vienna RNA dir is not provided.")
+
+  /** Parse and check any set of existing indices. */
+  private val ExistingIndices = existing match {
+    case None    => Seq.empty[Index]
+    case Some(p) => Io.readLines(p).map(_.trim.toUpperCase).filter(_.nonEmpty).map(_.getBytes).toSeq
+  }
+
+  if (ExistingIndices.exists(_.length != length)) invalid(s"Existing indices are not all of length ${length}")
+  if (ExistingIndices.exists(idx => idx.exists(b => !Bases.contains(b)))) invalid("One or more existing indices contains a non-[ACGT] character.")
+
+  /** The set of KMERs from the avoid sequences, that should not be selected. */
+  private val AvoidKmers = avoidSequence.flatMap { seq =>
+    (0 to seq.length - length).flatMap { i =>
+      val sub = seq.substring(i, i + length).toUpperCase
+      Seq(sub, SequenceUtil.reverseComplement(sub))
+    }
+  }.toSet
+
+  /* An Option[DnaFoldPredictor] to use to predict secondary structure. */
+  private val dnaFoldPredictor = viennaRnaDir.map(dir => new DnaFoldPredictor(dir.toFile, temperature))
+  private[util] val AdapterRegex = Pattern.compile("^[ACGT]*N+[ACGT]*$")
+  adapters.map(_.toUpperCase).filterNot(AdapterRegex.matcher(_).matches()) match {
+    case Seq() => Unit
+    case xs    => invalid(s"Adapters must be all [ACGT] with a single contiguous block of Ns where the index goes: ${xs}")
+  }
+
+  /** The main method! */
+  override def execute(): Unit = {
```

Comment:
It's not necessary, but I think it is good form, since it'll generate a compiler error if you use it and then mis-spell a method name, or change the name in the parent and miss during refactoring.